vpr/src/route/rr_graph_fpga_interchange.cpp

```diff
@@ -0,0 +1,1706 @@
+        }
+
+        for (auto const& i : NODESIDES) {
+            ends[i]->back_bone = true;
+            if (roots.find(ends[i]) != roots.end())
+                roots.erase(ends[i]);
+        }
+
+        bool right_to_top = ends[TOP_EDGE]->loc.first >= ends[BOTTOM_EDGE]->loc.first;
+        bool left_to_top = ends[TOP_EDGE]->loc.first < ends[BOTTOM_EDGE]->loc.first;
+
+        location right_comb_node(right_to_top ? ends[TOP_EDGE]->loc.first : ends[BOTTOM_EDGE]->loc.first, ends[RIGHT_EDGE]->loc.second);
+        location left_comb_node(left_to_top ? ends[TOP_EDGE]->loc.first : ends[BOTTOM_EDGE]->loc.first, ends[LEFT_EDGE]->loc.second);
+
+        add_comb_node(existing_nodes, ends, right_comb_node, right_to_top, RIGHT_EDGE, LEFT_EDGE);
+        add_comb_node(existing_nodes, ends, left_comb_node, left_to_top, LEFT_EDGE, RIGHT_EDGE);
+
+        int y = std::max(right_comb_node.second, left_comb_node.second);
+        location loc1(left_comb_node.first, y);
+        location loc2(right_comb_node.first, y);
+
+        intermediate_node* true_root = nullptr;
+
+        if (loc1 != left_comb_node && loc1 != loc2) {
+            true_root = create_final_root_node(existing_nodes, true, loc1, left_comb_node, right_comb_node, RIGHT_EDGE);
+        } else if (loc2 != right_comb_node && loc1 != loc2) {
+            true_root = create_final_root_node(existing_nodes, false, loc2, right_comb_node, left_comb_node, LEFT_EDGE);
+        } else if (loc1 != loc2) {
+            add_line(existing_nodes, loc1, loc2, {RIGHT_EDGE, LEFT_EDGE});
+            existing_nodes[loc1]->links[RIGHT_EDGE] = true;
+            existing_nodes[loc2]->links[LEFT_EDGE] = true;
+            true_root = existing_nodes[loc1];
+        } else {
+            loc1 = right_comb_node;
+            loc2 = left_comb_node;
+            if (right_comb_node.second > left_comb_node.second) {
+                std::swap(loc1, loc2);
+            }
+            add_line(existing_nodes, loc1, loc2, {TOP_EDGE, BOTTOM_EDGE});
+            existing_nodes[loc1]->links[TOP_EDGE] = true;
+            existing_nodes[loc2]->links[BOTTOM_EDGE] = true;
+            true_root = existing_nodes[loc1];
+        }
+
+        connect_dangling_roots(roots, existing_nodes, ends);
+
+        VTR_ASSERT(true_root != nullptr);
+        return true_root;
+    }
+
+    /*
+     * Removes dangling nodes from a graph represented by the root node.
+     * Dangling nodes are nodes that do not connect to a pin, a pip or other non-dangling node.
+     */
+    int reduce_graph_and_count_nodes_left(intermediate_node* root,
+                                          std::map<location, intermediate_node*>& existing_nodes,
+                                          int seg_id) {
+        int cnt = 0;
+        std::queue<intermediate_node*> walker;
+        std::stack<intermediate_node*> back_walker;
+        walker.push(root);
+        root->visited = root;
+        bool has_chanx;
```

Comment:
This might be uninitialized, leading to unexpected errors

Reply:
Fixed in next commit